torch/distributed/optim/named_optimizer.py

```diff
@@ -248,10 +239,9 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
                 param_keys.append(self.ordered_param_keys[param_key])  # type: ignore[call-overload]
             new_group_map[_gen_param_group_key(param_keys)] = new_group
         for group_key, new_group in new_group_map.items():
+            # When the conditional training is performed, not all parameters need to be in the param_group.
```

Comment:
I'd add some more detail here as "conditional training" is not really a well-known term IMO, maybe something like

"When not all parameters are used in training or receive gradient, not all parameters would be in the `param_group`. Thus we <whatever logic happens in this case>".

Reply:
Sure, thanks for the suggestion.